internal/pkg/client/cache/library.go

```diff
@@ -29,12 +32,25 @@ func LibraryImage(sum, name string) string {
 
 // LibraryImageExists returns whether the image with the SHA sum exists in the LibraryImage cache
 func LibraryImageExists(sum, name string) (bool, error) {
-	_, err := os.Stat(LibraryImage(sum, name))
+	imagePath := LibraryImage(sum, name)
+	_, err := os.Stat(imagePath)
 	if os.IsNotExist(err) {
 		return false, nil
 	} else if err != nil {
 		return false, err
 	}
 
+	cacheSum, err := client.ImageHash(imagePath)
+	if err != nil {
+		sylog.Debugf("Error getting ImageHash - Deleting Cached File: %v", err)
```

Comment:
This seems like an error that should be returned to the caller instead of just being part of debug output

Reply:
@ikaneshiro : whether failing to get ImageHash will fail the pull command itself?